src/lib/InstantSearch.ts

```diff
@@ -41,6 +41,11 @@ function defaultCreateURL() {
   return '#';
 }
 
+// this purposely breaks typescript's type inference to ensure it's not used
+// as it's used for a default parameter for example
+// source: https://github.com/Microsoft/TypeScript/issues/14829#issuecomment-504042546
+type NoInfer<T> = [T][T extends any ? 0 : never];
```

Comment:
Did you see this simplified version https://github.com/Microsoft/TypeScript/issues/14829#issuecomment-980401795 ?